api/src/controllers/users.ts

```diff
@@ -317,14 +317,51 @@ router.post(
 		});
 		await authService.verifyPassword(req.accountability.user, req.body.password);
 
-		const { url, secret } = await service.enableTFA(req.accountability.user);
+		const { url, secret } = await service.generateTFA(req.accountability.user);
 
 		res.locals.payload = { data: { secret, otpauth_url: url } };
 		return next();
 	}),
 	respond
 );
 
+router.post(
+	'/me/tfa/enable/',
+	asyncHandler(async (req, res, next) => {
+		if (!req.accountability?.user) {
+			throw new InvalidCredentialsException();
+		}
+
+		if (!req.body.secret) {
+			throw new InvalidPayloadException(`"secret" is required`);
+		}
+
+		if (!req.body.otp) {
+			throw new InvalidPayloadException(`"otp" is required`);
+		}
+
+		const service = new UsersService({
+			accountability: req.accountability,
+			schema: req.schema,
+		});
+
+		const authService = new AuthenticationService({
+			accountability: req.accountability,
+			schema: req.schema,
+		});
+		const otpValid = await authService.verifyOTP(req.accountability.user, req.body.otp, req.body.secret);
+
+		if (otpValid === false) {
+			throw new InvalidPayloadException(`"otp" is invalid`);
+		}
```

Comment:
Should we move this block of logic into the `enableTFA` method of the users service? It has to be duplicated between the different interfaces (eg REST / GraphQL) now 🤔

Reply:
Yes makes sense. I can change it tomorrow.

Reply:
I updated the code, so that the logic is within the enableTFA